modules/sdk-coin-dot/src/lib/iface.ts

```diff
@@ -142,12 +158,87 @@ export interface AddAnonymousProxyArgs {
  */
 export type BatchCallObject = {
   callIndex: string;
-  args: Record<string, any>;
+  args:
+    | Record<string, any>
+    | AddProxyBatchCallArgs
+    | AddAnonymousProxyBatchCallArgs
+    | StakeBatchCallArgs
+    | StakeMoreCallArgs;
 };
 export interface BatchArgs {
   calls: BatchCallObject[];
 }
 
+export interface AddAnonymousProxyBatchCallArgs {
+  proxy_type: ProxyType;
+  index: number;
+  delay: number;
+}
+
+export interface AddProxyBatchCallArgs {
+  delegate: string | AccountId;
+  proxy_type: ProxyType;
```

Comment:
Let's keep consistent with our naming in camelCase unless there is a specific reason it needs to be this weay.

Reply:
This is the shape of the object used by the underlying library we are using to decode transactions.

Reply:
A comment saying that ^ in the code would be helpful.

Reply:
Added a comment.